plugins/test/framework.cc

```diff
@@ -134,22 +195,47 @@ TestHttpContext::Result TestHttpContext::SendRequestHeaders(
   phase_logs_.clear();
   result_ = Result{.headers = std::move(headers)};
   phase_ = proxy_wasm::WasmHeaderMapType::RequestHeaders;
-  result_.status =
+  current_callback_ = TestHttpContext::CallbackType::RequestHeaders;
+  result_.header_status =
       onRequestHeaders(result_.headers.size(), /*end_of_stream=*/false);
   phase_ = proxy_wasm::WasmHeaderMapType(-1);  // ideally 0 would mean unset
   return std::move(result_);
 }
+
+TestHttpContext::Result TestHttpContext::SendRequestBody(std::string body) {
+  phase_logs_.clear();
+  result_ = Result{};
+  body_buffer_.set(std::move(body));
```

Comment:
IIUC this is calling `BufferBase::set(std::string_view data)`, which won't take ownership. I think you need to define a new `Buffer::SetOwned(std::string data)` method:

```
Buffer* SetOwned(std::string data) {
  clear();
  owned_string_buffer_ = std::move(data);
  return this;
}
```

I'd avoid naming this new method `set()` because then it will be unclear whether code is invoking `Buffer::set(std::string)` or `BufferBase::set(std::string_view)`.

Reply:
We don't want ownership at this point since there have been no body mutations. This lets us avoid unnecessary string copies in the case where there is no change.